enterprise/server/backends/authdb/authdb_test.go

```diff
@@ -0,0 +1,235 @@
+
+	akg, err := adb.GetAPIKeyGroupFromBasicAuth(ctx, g.GroupID, g.WriteToken)
+	require.NoError(t, err)
+
+	assert.Equal(t, randKey.GroupID, akg.GetGroupID())
+	assert.Equal(t, randKey.Capabilities, akg.GetCapabilities())
+	assert.Equal(t, false, akg.GetUseGroupOwnedExecutors())
+
+	// Using invalid or empty values should produce an error
+	akg, err = adb.GetAPIKeyGroupFromBasicAuth(ctx, "", "")
+	require.Nil(t, akg)
+	requireUnauthenticatedError(t, err)
+	akg, err = adb.GetAPIKeyGroupFromBasicAuth(ctx, "", g.WriteToken)
+	require.Nil(t, akg)
+	requireUnauthenticatedError(t, err)
+	akg, err = adb.GetAPIKeyGroupFromBasicAuth(ctx, g.GroupID, "")
+	require.Nil(t, akg)
+	requireUnauthenticatedError(t, err)
+	akg, err = adb.GetAPIKeyGroupFromBasicAuth(ctx, "INVALID", g.WriteToken)
+	require.Nil(t, akg)
+	requireUnauthenticatedError(t, err)
+	akg, err = adb.GetAPIKeyGroupFromBasicAuth(ctx, g.GroupID, "INVALID")
+	require.Nil(t, akg)
+	requireUnauthenticatedError(t, err)
+}
+
+func TestLookupUserFromSubID(t *testing.T) {
+	seed(t)
+	ctx := context.Background()
+	env := setupEnv(t)
+	adb := env.GetAuthDB()
+
+	users := enterprise_testauth.CreateRandomGroups(t, env)
+	randUser := users[rand.Intn(len(users))]
+
+	u, err := adb.LookupUserFromSubID(ctx, randUser.SubID)
+	require.NoError(t, err)
+	require.Equal(t, randUser, u)
+
+	// Using empty or invalid values should produce an error
+	u, err = adb.LookupUserFromSubID(ctx, "")
+	require.Nil(t, u)
+	requireRecordNotFoundError(t, err)
+	u, err = adb.LookupUserFromSubID(ctx, "INVALID")
+	require.Nil(t, u)
+	requireRecordNotFoundError(t, err)
+}
+
+func createRandomAPIKeys(t *testing.T, ctx context.Context, env environment.Env) []*tables.APIKey {
+	users := enterprise_testauth.CreateRandomGroups(t, env)
+	var allKeys []*tables.APIKey
+	// List the org API keys accessible to any admins we created
+	auth := env.GetAuthenticator().(*testauth.TestAuthenticator)
+	for _, u := range users {
+		authCtx, err := auth.WithAuthenticatedUser(ctx, u.UserID)
+		require.NoError(t, err)
+		if role.Role(u.Groups[0].Role) != role.Admin {
+			continue
+		}
+		keys, err := env.GetUserDB().GetAPIKeys(authCtx, u.Groups[0].Group.GroupID)
+		require.NoError(t, err)
+		allKeys = append(allKeys, keys...)
+	}
+	require.NotEmpty(t, allKeys, "sanity check: should have created some random API keys")
+	return allKeys
+}
+
+func setupEnv(t *testing.T) environment.Env {
+	env := enterprise_testenv.New(t)
+	enterprise_testauth.Configure(t, env) // provisions AuthDB and UserDB
+	return env
+}
+
+func seed(t *testing.T) {
+	seed := time.Now().UnixNano()
+	rand.Seed(seed)
+	t.Logf("Random seed: %d", seed)
+}
+
+func requireUnauthenticatedError(t *testing.T, err error) {
+	require.Truef(
+		t, status.IsUnauthenticatedError(err),
+		"expected Unauthenticated error; got: %v", err)
+}
+
+func requireRecordNotFoundError(t *testing.T, err error) {
+	require.Truef(
+		t, db.IsRecordNotFound(err),
+		"expected RecordNotFound error; got: %v", err)
+}
```

Comment:
These tests would be clearer without these methods -- they sort of obfuscate what's going on and don't really add a lot of additional value.

Reply:
Removed